core/service/app.go

```diff
@@ -628,6 +639,133 @@ func (a App) uploadAppPackageFiles(
 	return nil
 }
 
+type manifest struct {
+	Number         int      `yaml:"number"`
+	AppName        string   `yaml:"app_name"`
+	Description    string   `yaml:"description"`
+	HasUiExtension bool     `yaml:"has_ui_extension"`
+	Changes        []string `yaml:"changes"`
+	Prices         []struct {
+		Currency entity.Currency `yaml:"currency"`
+		Amount   int             `yaml:"amount"`
+	}
+}
+
+type Uploader func(io.Reader) *errs.Error
+
+/*
+sanitizeTarHeaderName removes the first directory name from the header name.
+The file extracted from tar will have the name of tar as prefix of the file name.
+The prefix is defined by the user
+Eg:
+tar name: app.tar
+before tar: manifest.yaml
+after tar: app/manifest.yaml
+*/
+func (a App) sanitizeTarHeaderName(headerName string) string {
+	headerName = tarNameSanitizer.ReplaceAllString(headerName, "")
+
+	return headerName
+}
+
+func (a App) processFile(ct context.Context, userID uint64, appID uint64, reader *tar.Reader, fileName string, uploader Uploader) *errs.Error {
+	switch fileName {
+	case "manifest.yaml":
+		return a.processManifestFile(ct, userID, appID, reader, uploader)
+	default:
+		return uploader(reader)
+	}
+}
+
+func (a App) processManifestFile(ct context.Context, userID uint64, appID uint64, reader *tar.Reader, uploader Uploader) *errs.Error {
+	manifestData := manifest{}
+	multiReader := tmio.NewMultiReaders()
+	readers := multiReader.GenerateMultiReaders(reader, 2)
```

Comment:
```suggestion
	readers := tmio.NewMultiReaders(reader, 2)
```

Is the additional `GenerateMultiReaders` necessary? Can we merge `NewMultiReaders` and `GenerateMultiReaders` to a single API?